dataprofiler/profilers/helpers/report_helpers.py

```diff
@@ -100,7 +118,50 @@ def _prepare_report(report, output_format=None, omit_keys=None):
         if isinstance(value, set):
             value = sorted(list(value))
 
-        if isinstance(value, dict):
+        # For data_stats (in structured case), need to recurse through a list
+        # As well as account for omit_keys containing indices instead of keys
+        if key == "data_stats" and isinstance(value, list):
+            fmt_report["data_stats"] = []
```

Comment:
any reason we cannot format in place? why do we need to create a new list and replace?

Reply:
I see bc we want to remove from. format. Another option is to keep a list of indexes to keep and subset after (only fix those keeping)